src/util-buffer.h

```diff
@@ -25,9 +25,9 @@
 #define __UTIL_BUFFER_H__
 
 typedef struct MemBuffer_ {
-   uint8_t *buffer;
    uint32_t size;
    uint32_t offset;
+   uint8_t buffer[0];
```

Comment:
So this shrinks it from 16 to 9 bytes? Looks like this might not be worth it if the memory block following this header is now unaligned?